core/store/src/cold_storage.rs

```diff
@@ -239,6 +249,20 @@ where
 
 #[allow(dead_code)]
 impl StoreWithCache<'_> {
+    pub fn iter_prefix_with_callback(
+        &mut self,
+        col: DBCol,
+        key_prefix: &[u8],
+        mut callback: impl FnMut(Box<[u8]>),
+    ) -> io::Result<()> {
+        for iter_result in self.store.iter_prefix(col, key_prefix) {
+            let (key, value) = iter_result?;
+            self.cache.insert((col.clone(), key.to_vec()), Some(value.to_vec()));
```

Comment:
```suggestion
            self.cache.insert((col, key.to_vec()), Some(value.into()));
```
DBCol is Copy so no need to Clone.  Furthermore, there’s `Form<Box<T>>` implementation for `Vec<T>` which avoids copying so better to use into for value.